swap/src/cli/command.rs

```diff
@@ -48,6 +50,9 @@ pub enum Command {
 
         #[structopt(flatten)]
         monero_params: MoneroParams,
+
+        #[structopt(long = "tor-socks5-port", help = "Your local Tor socks5 proxy port", default_value = DEFAULT_TOR_SOCKS5_PORT)]
+        tor_socks5_port: u16,
```

Comment:
He can, if the the port is not reachable, we send everything via clearnet. 
Maybe the an optional field ?

Reply:
> He can, if the the port is not reachable, we send everything via clearnet.

That is what I meant with "providing an invalid port" and I think it is quite weird that I have to choose and invalid port to disable Tor. Ideally, I can say `--no-tor` or something. I didn't want to get into the details of actually designing the CLI here just pointing out the lack of a clear "no tor" option.

Reply:
That's a hard one :D 

I played around with it a bit way too long and think the current solution is the best one. 

The issue is that if `alice_multiaddr` is an onion address, we need to have a `socks5_port`. 
If this can be expressed with `structopt`, that'd be awesome. 

- 
I wasn't able to achieve this, so I have the issue that if `tor_socks5_port` is an option and not provided, we will fail eventually when trying to dial to Alice. 

If it is not an option and defaults to a default port, we can fall back to clear net if Tor is not available (current implementation). 

So what I'm doing now is I provide a default port and fail Swap early if an onion address was provided but Tor was not running on the default port. 

This seem to me the best trade-off, e.g. a user who has Tor running on default port can simple do `swap buy-xmr -seller /onion/...` and the the communication will go through the default port. 

Check the latest commit for some details: #1a1b6ae